### PR TITLE
[4.1.0 migration] Check SOAP API validation in 2.6.0-4.1.0 pre validation step

### PR DESCRIPTION
**Pull request by msm1992**
## Purpose
- Fixes wsdl pre-validation step for SOAP and SOAPTOREST from 2.6.0

## Approach
- Pre-validation step is executed before running the APIM migration client. Hence the 2.6.0 SOAP and SOAPTOREST APIs would still have their type set as HTTP.
- Due to that WSDL validation for those APIs are skipped.
- As the fix, added a util method to figure out correct API Type based on overview_wsdl field and soap_to_rest registry resource. (Same logic used in PR https://github.com/wso2-extensions/apim-migration-resources/pull/214)

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: components/migration-client/wso2-api-migration-client/src/main/java/org/wso2/carbon/apimgt/migration/validator/validators/Validator.java [hunk 1]
```diff
@@ -32,7 +33,16 @@ public void validate(UserRegistry registry, GenericArtifact artifact, String pre
         this.apiName = artifact.getAttribute(APIConstants.API_OVERVIEW_NAME);
         this.apiVersion = artifact.getAttribute(APIConstants.API_OVERVIEW_VERSION);
         this.provider = artifact.getAttribute(APIConstants.API_OVERVIEW_PROVIDER);
-        this.apiType = artifact.getAttribute(APIConstants.API_OVERVIEW_TYPE);
+
+        // At this point of  pre-validation step, SOAP and SOAPTOREST APIs from 2.6.0 will have their overview_type
+        // set as HTTP, hence we are employing a Util method to fetch correct API Type based on other resources and
+        // artifact fields.
+        if ("2.6.0".equals(utils.getMigrateFromVersion())) {
```
Comment (HiranyaKavishani) [inline components/migration-client/wso2-api-migration-client/src/main/java/org/wso2/carbon/apimgt/migration/validator/validators/Validator.java:40]: Shall we make "2.6.0" as a constant?

File: components/migration-client/wso2-api-migration-client/src/main/java/org/wso2/carbon/apimgt/migration/validator/utils/V260Utils.java [hunk 2]
```diff
@@ -26,4 +35,30 @@ public String getWSDLPath(String apiName, String apiVersion, String provider) {
...
+            try {
+                if (SOAPOperationBindingUtils.isSOAPToRESTApi(artifact.getAttribute(Constants.API_OVERVIEW_NAME),
+                        artifact.getAttribute(Constants.API_OVERVIEW_VERSION),
+                        artifact.getAttribute(Constants.API_OVERVIEW_PROVIDER))) {
+                    apiType = Constants.API_TYPE_SOAPTOREST;
+                } else {
+                    apiType = Constants.API_TYPE_SOAP;
+                }
+            } catch (APIManagementException e) {
+                log.error("Error occurred when getting artifact manager", e);
```
Comment (HiranyaKavishani) [inline components/migration-client/wso2-api-migration-client/src/main/java/org/wso2/carbon/apimgt/migration/validator/utils/V260Utils.java:59]: log.error("Error occurred when getting attributes from artifact manager ", e);

File: components/migration-client/wso2-api-migration-client/src/main/java/org/wso2/carbon/apimgt/migration/validator/validators/Validator.java (hunk 1 above)
Comment (msm1992) [reply components/migration-client/wso2-api-migration-client/src/main/java/org/wso2/carbon/apimgt/migration/validator/validators/Validator.java]: fixed in https://github.com/wso2-extensions/apim-migration-resources/pull/216/commits/3accab6fcc72e95f16389142fea122677f584323

File: components/migration-client/wso2-api-migration-client/src/main/java/org/wso2/carbon/apimgt/migration/validator/utils/V260Utils.java (hunk 2 above)
Comment (msm1992) [reply components/migration-client/wso2-api-migration-client/src/main/java/org/wso2/carbon/apimgt/migration/validator/utils/V260Utils.java]: fixed in https://github.com/wso2-extensions/apim-migration-resources/pull/216/commits/3accab6fcc72e95f16389142fea122677f584323

**Pull request merged**